docs/src/userguide/config/event_config.rst

```diff
@@ -126,13 +126,176 @@ Example configuration for a custom event input that is not supported by Klio:
 
     Any arbitrary key-value pairs for custom event input configuration specific to a job.
 
+.. _event-config-bigquery:
 
-.. todo:: document supported batch event inputs
+GoogleBigQuery
+^^^^^^^^^^^^^^
+
+*Mode: Batch*
+
+Consuming KlioMessages from `Google BigQuery`_ is supported for all runners. Klio allows the user to specify events from queries and from tables.
+
+Example configuration for `Google BigQuery`_, by specifying a table:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              columns:
+              - entity_id
+              dataset: bigquery_dataset
+              project: gcp_project
+              table: bigquery_table
 
 
+Example configuration for `Google BigQuery`_, by specifying a query in the BigQuery dialect:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              query:
+                SELECT * FROM [gcp_project.bigquery_dataset.bigquery_table]
+
+.. option:: job_config.events.inputs[].type
+
+    Value: ``bq``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.inputs[].columns[]
+
+    Not applicable if query is specified.
+
+    A list of strings specifying the columns to read events from.
+    If only one column is provided, the value will be returned as a bytestring.
+    If no columns are specified, or if more than one column is specified,
+    the results including the column names will be serialized to a JSON bytestring,
+    e.g. ``'{"field1": "foo", "field2": bar"}'``.
+
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].columns[].<column> STR
+
+    A column in the table or query result used to build the event input from.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].table STR
+
+    Name of the event input table.
+
+    If specified all data of the table will be used as input of the current source.
+    The ID must contain only letters ``a-z``, ``A-Z``, numbers ``0-9``,
+    or underscores ``_``.
+
+    If dataset and query arguments are not specified,  then the table argument must
+    contain the entire table reference specified as:
+    ``'DATASET.TABLE'`` or ``'PROJECT:DATASET.TABLE'``.
+
+    | **Runner**: Dataflow, Direct
+    | *Required unless query provided*
```

Comment:
Isn't it (as well as `project`/`dataset`) mutually exclusive with `query` as well?

Reply:
Ah, is the goal here to also mention that specifying `project/dataset/table` and `query` are mutually exclusive? I felt that since it was mentioned in other places, it wasn't necessary here, but I can repeat it if it would be clearer.

Reply:
Ah I see, you want me to also make sure project/dataset aren't included. I was trying to be terse since you can't really specify a table by itself without the dataset and project, but I can go through and make it explicit.

Reply:
yeah it was a bit of both. If `query` is provided, then `project` + `dataset` + `table` shouldn't be given, and vice-versa. And even if it is repetitive, to still mention it because I could very well see someone read this documentation on `table` and not even look at `project`, `dataset`, or `query`.  Maybe a user ends up providing `PROJECT:DATASET.TABLE` for the `table` value, not realizing that they should use all three fields. Then they add the `query` and assume they're needing to essentially define the full table path twice.

Goodness I can't wait until we implement config validation 😆